### PR TITLE
Make es-ES to fall back to es-MX if the translation doesn't exist

### PR DESCRIPTION
**Pull request by bethanyaconnor**
For a user in es-ES, we will first look for an es-ES translation then an es-MX one before falling back to en like normal. Unfortunately, just setting this in config.i18n.fallbacks didn't work. If there is a better place to put this, let me know. This does feel a bit weird but as long we don't have complicated fallback rules, I think it's probably okay.

Useful documentation:
- https://github.com/ruby-i18n/i18n/blob/ba83938beb6f9a042ac450c9fd3253de0c9fb6b1/lib/i18n/locale/fallbacks.rb#L39
http://pawelgoscicki.com/archives/2015/02/enabling-i18n-locale-fallbacks-in-rails/

Current production in es-ES for studio.code.org/s/coursec-2017:
Screenshot 2019-07-03 at 12 18 57 PM

With this change:
Screenshot 2019-07-03 at 12 18 29 PM

### CODE REVIEW

File: dashboard/config/application.rb [hunk 1]
```diff
@@ -102,6 +102,10 @@ class Application < Rails::Application
       end
     end
 
+    config.after_initialize do
```
Comment (Hamms) [inline dashboard/config/application.rb:105]: why is this necessary?

Comment (bethanyaconnor) [reply dashboard/config/application.rb:105]: The fallbacks seem to be overwritten at some point after config initialization. 
I found this at http://pawelgoscicki.com/archives/2015/02/enabling-i18n-locale-fallbacks-in-rails/ (linked in the description).

Comment (Hamms) [reply dashboard/config/application.rb:105]: Is that also true for `config.i18n.fallbacks[:defaults] = ['en-US']` above?

Comment (bethanyaconnor) [reply dashboard/config/application.rb:105]: The default locale seems to work a bit differently for some reason. Even the [docs](https://github.com/ruby-i18n/i18n/wiki/Fallbacks) linked by the i18n rails gem has custom fallbacks modifying `I18n.fallbacks` instead of `config.i18n.fallback`.

If there's a better place to put this, I'm happy to move it there but I don't think it can be specified in `config.i18n`.

Comment (Hamms) [reply dashboard/config/application.rb:105]: Weird. This seems good, then, but would you add a comment explaining why we're doing it in this file and in the block? Possibly add one or both of those links

**Review by Hamms — APPROVED** [review]
